src/crypto/poly1305.h

```diff
@@ -32,6 +35,40 @@ void poly1305_finish(poly1305_context *st, unsigned char mac[16]) noexcept;
 
 }  // namespace poly1305_donna
 
+/** C++ wrapper with std::byte Span interface around poly1305_donna code. */
+class Poly1305
+{
+    poly1305_donna::poly1305_context m_ctx;
+
+public:
+    /** Length of the output produced by Finalize(). */
+    static constexpr unsigned TAGLEN = POLY1305_TAGLEN;
+
+    /** Length of the keys expected by the constructor. */
+    static constexpr unsigned KEYLEN = POLY1305_KEYLEN;
+
+    /** Construct a Poly1305 object with a given 32-byte key. */
+    Poly1305(Span<const std::byte> key) noexcept
+    {
+        assert(key.size() == KEYLEN);
+        poly1305_donna::poly1305_init(&m_ctx, reinterpret_cast<const unsigned char*>(key.data()));
```

Comment:
In 7601a60cddc4405f58f71879f996322329509dfe "crypto: add Poly1305 class with std::byte Span interface"

Use `UCharCast` cast here (and below)?